app/models/status.rb

```diff
@@ -361,6 +368,7 @@ def permitted_for(target_account, account)
         scope = left_outer_joins(:reblog)
 
         scope.where(visibility: visibility)
+             .not_hidden(account)
```

Comment:
This is going to change a possibly expensive request, which we have an index tailored for. We should make sure this doesn't affect performances badly.